library/havege.c

```diff
@@ -38,6 +38,7 @@
 #include "mbedtls/timing.h"
 #include "mbedtls/platform_util.h"
 
+#include <stdint.h>
```

Comment:
Style: Should we bother to include `stdint.h` again as we'd have already included it via including `havege.h`?

Reply:
I tend to prefer to include headers for what they define, not for what headers they pull in. According to this style, since `havege.c` uses `uint32_t`, it includes `stdint.h`.

Reply:
Is this the convention we use in the rest of Mbed TLS as well? I'm happy to use whatever the library already does.

It's slightly more efficient to not `#include` a header if you know the types are already available, as they must since the declarations we use from `havege.h` require it, because the preprocessor doesn't need to read in that file at all. For large projects, and especially on slow filesystems (e.g. Windows, network drives), this can add up to a big savings in compile time. It's not uncommon to have a style that encourages efficiency. (C++ style guides recommending pre-increment operators over post-increment operators in for loops is one example of this.)

Reply:
I don't remember that we ever discussed this, and I don't think we're doing or not-doing this with any noticeable consistency.

All our header files have garden-variety include guards. Compilers typically optimize well for that case. In any case, compilation time is the least of our worries. This is not a 100000-file C++ project. Being a security project, we value security, readability and ease of maintenance rather more than runtime performance, let alone build-time performance. Also, if you want to optimize compilation time, the biggest thing by far would be to get rid of cmake!